### ISSUE TITLE
Allow parsing of TRANSFAC matrices where the first matrix line starts with 00 instead of 01 (or not)

### ISSUE THREAD

**Issue by ghuls** [biopython/biopython#550]
Allow parsing of TRANSFAC matrices where the first matrix line starts with 00 instead of 01 (or not)?

Some TRANSFAC matrices on the net start the matrix line with 00 instead of 01.

According to http://meme-suite.org/doc/transfac-format.html it does not seem to be valid, but
those matrices still exist. So would it be acceptable to support those "invalid" TRANSFAC files?

Example:

"Motif Weight Matrices" for Saccharomyces cerevisiae from http://swissregulon.unibas.ch/fcgi/sr/downloads has motif "SIP4" in this format.

Direct link: http://swissregulon.unibas.ch/ozonov/wms.txt

```
NA    SIP4
PO    A    C    G    T
00    0.00    14.20    0.00    0.00
01    0.00    0.00    14.20    0.00
02    0.32    0.00    13.88    0.00
03    9.65    2.77    2.11    0.00
04    0.00    3.09    0.00    11.11
05    1.75    0.00    10.74    1.88
06    14.20    0.00    0.00    0.00
07    13.88    0.00    0.00    0.00
08    7.11    0.00    0.00    7.60
09    0.00    0.00    11.59    2.29
10    0.00    0.00    13.88    0.00
11    12.15    0.00    2.28    0.00
12    5.13    2.45    6.98    0.16
//
```

This patch would allow supporting this invalid format, while still checking that the numbering of the matrix goes up by 1 each time the counts are added to counts:

``` diff
diff --git a/Bio/motifs/transfac.py b/Bio/motifs/transfac.py
index 4b032b7..8377966 100644
--- a/Bio/motifs/transfac.py
+++ b/Bio/motifs/transfac.py
@@ -107,7 +107,11 @@ def read(handle):
                     i = int(key)
                 except ValueError:
                     break
-                length += 1
+                # Support also invalid TRANSFAC files which start the matrix with "00" instead of "01".
+                if length == 0 and i == 0:
+                    pass
+                else:
+                    length += 1
                 assert i == length
                 values = value.split()
                 for c, v in zip("ACGT", values):
```

If this kind of change is accepted, I can add it to:
https://github.com/biopython/biopython/pull/549

**Comment by peterjc** [comment]
Following the discussion on #549, my recommendation would be we apply something based on this change but where the "odd" ``00`` lines give a warning, e.g.:

``` python
import warnings
from Bio import BiopythonParserWarning
...
warnings.warn("TRANSFAC matrices should start with 01, but this file uses 00 instead.",
              BiopythonParserWarning)
```

**Comment by peterjc** [comment]
Closed via #1018
